front_end_pocket_poll/lib/groups_widgets/group_page.dart

```diff
@@ -41,6 +43,45 @@ class _GroupPageState extends State<GroupPage>
   final int votingTab = 2;
   final int occurringTab = 1;
   final int closedTab = 4;
+  final Map<int, int> listIndexesToEventTypes = new Map<int, int>();
+
+  //set up all of the variables needed for tracking the event lists
+  final Map<int, int> eventTypesToLargestBatchIndexLoaded = {
+    EventsList.eventsTypeNew: 0,
+    EventsList.eventsTypeVoting: 0,
+    EventsList.eventsTypeConsider: 0,
+    EventsList.eventsTypeClosed: 0,
+    EventsList.eventsTypeOccurring: 0
+  };
+
+  //this 'batch limit' is the index of the first batch that comes back empty
+  final Map<int, int> eventTypesToBatchLimits = {
+    EventsList.eventsTypeNew: null,
+    EventsList.eventsTypeVoting: null,
+    EventsList.eventsTypeConsider: null,
+    EventsList.eventsTypeClosed: null,
+    EventsList.eventsTypeOccurring: null
+  };
+
+  //this 'batch limit' is the index of the first batch that comes back empty
+  final Map<int, Map<int, List<String>>> eventTypesToBatchEventIds = {
+    EventsList.eventsTypeNew: new Map<int, List<String>>(),
+    EventsList.eventsTypeVoting: new Map<int, List<String>>(),
+    EventsList.eventsTypeConsider: new Map<int, List<String>>(),
+    EventsList.eventsTypeClosed: new Map<int, List<String>>(),
+    EventsList.eventsTypeOccurring: new Map<int, List<String>>()
+  };
+
+  final Map<int, double> eventTypesToPreviousMaxScrollExtents = {
+    EventsList.eventsTypeNew: null,
+    EventsList.eventsTypeVoting: null,
+    EventsList.eventsTypeConsider: null,
+    EventsList.eventsTypeClosed: null,
+    EventsList.eventsTypeOccurring: null
+  };
+
+  final Map<int, double> eventListScrollPositions = {};
```

Comment:
I would add a comment here to explain what this does (like what the mapping is, I'm pretty sure it's eventType to offset, but would be good to have it said explicitly)

Reply:
added comment